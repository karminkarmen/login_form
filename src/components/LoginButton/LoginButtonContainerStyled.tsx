import styled from 'styled-components';
import { centered } from '../../configs/mixins';
import { innerWidthSize } from '../../constants/themeConsts';

const LoginButtonContainerStyled = styled.div`
    ${centered}
    position: relative;
    width: ${innerWidthSize / 2}px;
    height: ${innerWidthSize / 2}px;
    left: ${innerWidthSize + 10}px;
    top: -130px;
    border-radius: ${props => props.theme.radius.maximum};
    background: ${props => props.theme.colors.main};
    transition: ${props => props.theme.transitionTime.default};

    &:hover {
        background-color: ${props => props.theme.colors.secondaryLighten};
        cursor: pointer;
    }

    &:focus {
        outline: none;
    }
`;

export default LoginButtonContainerStyled;
