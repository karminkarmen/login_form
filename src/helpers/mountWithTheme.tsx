import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../configs/theme';

const mountWithTheme = (child: React.ReactElement) => {
    return mount(child, {
        wrappingComponent: ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    });
}

export default mountWithTheme;