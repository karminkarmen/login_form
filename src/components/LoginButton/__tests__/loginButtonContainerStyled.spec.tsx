import React from 'react';
import mountWithTheme from '../../../helpers/mountWithTheme';
import LoginButtonContainerStyled from '../LoginButtonContainerStyled';

describe('<LoginButtonContainerStyled />', () => {

    it('matches snapshot', () => {
        const wrapper = mountWithTheme(<LoginButtonContainerStyled />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

});
