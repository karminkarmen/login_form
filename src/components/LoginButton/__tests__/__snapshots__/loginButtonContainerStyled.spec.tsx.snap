// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButtonContainerStyled /> matches snapshot 1`] = `"<div class=\\"sc-AxjAm gxQmdq\\"></div>"`;
